### DISCUSSION TITLE
Ability to specify a different `meson.build` file

### DISCUSSION THREAD

**Discussion by LunarLambda** [mesonbuild/meson#12777] [Ideas]
I have a project with a large number of `subprojects` (essentially a mini wrap-DB for a very specific platform). For development I currently use a git branch with a different `meson.build` file that just compiles every single subproject instead of only the ones used as dependencies. This has some downsides, like not being able to properly `git merge` and needing to manually import the `subprojects/` directory, cross files, etc. to the main branch. It's also suboptimal because release tags are done against the development branch. Using a subdirectory with a symlink to `subprojects` is a pretty ugly hack in my opinion. It would be nice if `meson setup` just let me specify a different filename, say `meson.build.dev` or whatever.

[The project](https://github.com/LunarLambda/meson-gba) [Branch diff](https://github.com/LunarLambda/meson-gba/compare/main..dev)

**Comment by eli-schwartz** [comment]
It seems like it would be a lot easier to just have both versions in `meson.build` but separated by an if/else, and then you could consistently `git merge` the changes without fear. All you'd need to do is ensure that a single global variable defining whether or not you are in dev mode, is unchanged and different between branches -- since that section of the file wouldn't change, `git merge` wouldn't create conflict markers for it.

**Reply by LunarLambda** [reply]
yeah. I wanted to keep the user-facing "template" build code separate from the dev code. Maybe I could do a subdir call and call that conditionally instead though. That would keep it small. I'll try that

**Reply by LunarLambda** [reply]
Added `meson/dev` subdir similar to `meson/toolchain` and then the following

```python
if get_option('dev')
    subdir('meson/dev')
    subdir_done()
endif
```
